homeassistant/loader.py

```diff
@@ -883,11 +888,7 @@ async def async_get_integrations(
     hass: HomeAssistant, domains: Iterable[str]
 ) -> dict[str, Integration | Exception]:
     """Get integrations."""
-    if (cache := hass.data.get(DATA_INTEGRATIONS)) is None:
-        if not _async_mount_config_dir(hass):
-            return {domain: IntegrationNotFound(domain) for domain in domains}
```

Comment:
Do we want to keep this? We could remember if `_async_mount_config_dir` failed and bail out early here if it did?

Reply:
This is weird. `_async_mount_config_dir` should raise if it's called and no config dir is set.

Reply:
OK, but what should happen then?

Reply:
We should just not allow config dir to be `None` 🤷 It doesn't happen when we run HA and it doesn't run in tests. Without a config dir we cannot even write things like entity registry etc, nothing will work.

Reply:
OK, sounds good. I'll clean that up in a separate PR.